File: java/src/main/java/net/razorvine/pickle/objects/SetConstructor.java
package net.razorvine.pickle.objects;

import net.razorvine.pickle.IObjectConstructor;

import java.util.ArrayList;
import java.util.HashSet;

/**
 * This object constructor creates sets.
 *
 * @author Irmen de Jong (irmen@razorvine.net)
 */
public class SetConstructor implements IObjectConstructor {

	public SetConstructor() {
	}

	public Object construct(Object[] args) {
		// create a HashSet, args=arraylist of stuff to put in it
		@SuppressWarnings("unchecked")
		ArrayList<Object> data = (ArrayList<Object>) args[0];
		return new HashSet<>(data);
	}
}

File: dotnet/Razorvine.Pickle/Pickle/Pickle/Objects/SetConstructor.cs
/* part of Pickle, by Irmen de Jong (irmen@razorvine.net) */

using System.Collections;
using System.Collections.Generic;

namespace Razorvine.Pickle.Objects
{

/// <summary>
/// This object constructor creates sets. (HashSet&lt;object&gt;)
/// </summary>
public class SetConstructor : IObjectConstructor {

	public object construct(object[] args) {
		// create a HashSet, args=arraylist of stuff to put in it
		var elements=(ArrayList)args[0];
		return new HashSet<object>(elements.ToArray());
	}
}

}
